# Electronics Store dollar/hour analysis

## tl;dr

当前瞬时最高是 Q1 Laptops；若要求同价位至少 1,000 件，当前最高是 Q7 Smart Phones。结果均按每一级 Electronics Store 表示。

## Context & Methods

The notebook combines a timestamped Exchange order-book snapshot with the current economy-state-1 retail equation extracted from the live game bundle. For every product and visible quality tier, it searches retail prices in $0.01 increments and deducts the current administration-adjusted Electronics Store wage. Exchange buyers pay neither the seller's 4% fee nor seller-side Transport.

### Key assumptions

- Results are per Electronics Store level; throughput and wages both scale with store level while administration overhead is held at its current value.
- The stable winner requires at least 1,000 units at the quoted best price for that quality.
- Exchange prices are instantaneous and retail saturation updates over time.


In [1]:
from analysis import build_analysis
analysis = build_analysis()
print(analysis['best_by_product'])


[
  {
    "kind": 26,
    "product": "Laptops",
    "quality": 1,
    "buy_price": 1170.0,
    "offer_depth": 66,
    "optimal_retail_price": 1582.04,
    "gross_spread_per_unit": 412.04,
    "units_per_hour_per_store_level": 0.847367,
    "wage_per_hour_per_store_level": 182.647059,
    "net_profit_per_hour_per_store_level": 166.502214,
    "depth_hours_per_store_level": 77.89,
    "buyer_exchange_fee": 0.0,
    "buyer_transport_cost": 0.0,
    "rank": 1,
    "product_rank": 1
  },
  {
    "kind": 98,
    "product": "Quadcopters",
    "quality": 3,
    "buy_price": 870.0,
    "offer_depth": 296,
    "optimal_retail_price": 1304.65,
    "gross_spread_per_unit": 434.65,
    "units_per_hour_per_store_level": 0.777706,
    "wage_per_hour_per_store_level": 182.647059,
    "net_profit_per_hour_per_store_level": 155.383043,
    "depth_hours_per_store_level": 380.61,
    "buyer_exchange_fee": 0.0,
    "buyer_transport_cost": 0.0,
    "rank": 2,
    "product_rank": 2
  },
  {
    "kind": 24,
 

## Results

The spot winner benefits from a cheap but shallow Q1 Laptop offer. Q7 Smart Phones trade a small amount of hourly profit for much deeper sourcing, making them the more robust continuous-arbitrage choice.


In [2]:
print(analysis['quality_checks'])


{
  "products_covered": 6,
  "quality_offers_evaluated": 46,
  "all_products_have_a_winner": true,
  "all_results_are_positive": true,
  "spot_winner": "Laptops Q1",
  "stable_winner_minimum_depth_1000": "Smart Phones Q7",
  "buyer_fee_and_transport_are_zero": true,
  "one_cent_price_grid": true
}


## Takeaways

- Buy Q1 Laptops only while the $1,170 offer remains available.
- Use Q7 Smart Phones as the current deep-liquidity default.
- Recalculate immediately before buying because the ranking is quote-sensitive.
